src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -96,20 +120,44 @@ public class DocumentReplicationRequestHandler : DatabaseRequestHandler
         {
             switch (messageTypeAsString)
             {
+				case Constants.Replication.MessageTypes.Heartbeat:
+					_lastHeartbeatReceive = DateTime.UtcNow; //not sure that this variable is actually needed
+		            _heartbeatTimer.Change(TimeSpan.Zero, HeartbeatTimeout);
+					break;
                 case Constants.Replication.MessageTypes.GetLastEtag:
-                    using (context.OpenReadTransaction())
-                        WriteLastEtagResponse(webSocketStream, context, srcDbId);
-                    break;
+		            try
+		            {
+			            using (context.OpenReadTransaction())
+				            WriteLastEtagResponse(webSocketStream, context, srcDbId);
+		            }
+		            catch (Exception e)
+		            {
+			            Log.Error("Failed to write last etag response; Reason: " + e);
+		            }
+		            break;
                 case Constants.Replication.MessageTypes.ReplicationBatch:
                     BlittableJsonReaderArray replicatedDocs;
                     if (!message.TryGet(Constants.Replication.PropertyNames.ReplicationBatch, out replicatedDocs))
                         throw new InvalidDataException(
-                            $"Expected the message to have a field with replicated document array, named {Constants.Replication.PropertyNames.ReplicationBatch}. The property wasn't found");
-                    using (context.OpenWriteTransaction())
-                    {
-                        incomingReplication.ReceiveDocuments(context, replicatedDocs);
-                        context.Transaction.Commit();
-                    }
+                            $"Expected the message to have a field with replicated document array, named {Constants.Replication.PropertyNames.ReplicationBatch}. The property wasn't found");		           
+
+		            try
+					{										
+						using (context.OpenWriteTransaction())
+			            {
+				            incomingReplication.ReceiveDocuments(context, replicatedDocs);
+				            context.Transaction.Commit();
+			            }
+		            }
+		            catch (Exception e)
+		            {
+						Log.Error($"Received replication batch with {replicatedDocs.Length} documents, but failed to write it locally. Reason : {e}");
```

Comment:
Nope, should error and then close connection with message.
If this is a transient error the other side will reconnect to us.